File: DAY25.c/ques97.py
#write a program to merge two sorted arrays
def merge_sorted_arrays(arr1, arr2):
    merged_array = []  # List to store the merged result
    i, j = 0, 0  # Pointers for arr1 and arr2
    
    # Iterate through both arrays until one is exhausted
    while i < len(arr1) and j < len(arr2):
        if arr1[i] < arr2[j]:
            merged_array.append(arr1[i])  # Append smaller element from arr1
            i += 1  # Move pointer in arr1
        else:
            merged_array.append(arr2[j])  # Append smaller element from arr2
            j += 1  # Move pointer in arr2
            
    # If there are remaining elements in arr1, append them
    while i < len(arr1):
        merged_array.append(arr1[i])
        i += 1
        
    # If there are remaining elements in arr2, append them
    while j < len(arr2):
        merged_array.append(arr2[j])
        j += 1
        
    return merged_array  # Return the merged sorted array

File: DAY25.c/ques97.c
//write a program to merge two sorted arrays
#include <stdio.h>

int main() {
    int arr1[100], arr2[100];
    int n1, n2;
    printf("Enter the number of elements in the first array: ");
    scanf("%d", &n1);
    printf("Enter the elements of the first array (sorted): ");
    for (int i = 0; i < n1; i++) {
        scanf("%d", &arr1[i]);
    }
    printf("Enter the number of elements in the second array: ");
    scanf("%d", &n2);
    printf("Enter the elements of the second array (sorted): ");
    for (int i = 0; i < n2; i++) {
        scanf("%d", &arr2[i]);
    }
    int merged[n1 + n2];
    int i = 0, j = 0, k = 0;

    while (i < n1 && j < n2) {
        if (arr1[i] < arr2[j]) {
            merged[k++] = arr1[i++];
        } else {
            merged[k++] = arr2[j++];
        }
    }

    while (i < n1) {
        merged[k++] = arr1[i++];
    }

    while (j < n2) {
        merged[k++] = arr2[j++];
    }

    printf("Merged array: ");
    for (int l = 0; l < n1 + n2; l++) {
        printf("%d ", merged[l]);
    }
    printf("\n");

    return 0;
}